MoreLinq/Traverse.cs

```diff
@@ -88,19 +78,30 @@ public static IEnumerable<T> TraverseDepthFirst<T>(T root, Func<T, IEnumerable<T
         {
             if (childrenSelector == null) throw new ArgumentNullException(nameof(childrenSelector));
 
+            // because a stack pops the elements out in LIFO order, we need to push them in reverse
+            // if we want to traverse the returned list in the same order as was returned to us
+
+            var stack = new Stack<T>();
+            return TraverseImpl(root, x => childrenSelector(x).Reverse(), stack.Push, stack.Pop, stack);
```

Comment:
We can avoid the reverse by using a stack (or it is a queue), of children enumerator. We should take care of the dispose.

Reply:
Maybe that can be a separate change.